lua/ttt2/extensions/util.lua

```diff
@@ -454,6 +454,40 @@ end
 -- @function util.Capitalize(str)
 util.Capitalize = string.Capitalize
 
+---
+-- Checks if the provided team is an evil team. By default all non innocents that aren't
+-- `TEAM_NONE` are included. Can be disabled with the convar `ttt2_rolecheck_all_evil_roles`.
+-- @param string team The team that should be tested
+-- @return boolean Returns if a team is evil
+-- @realm shared
+function util.IsEvilTeam(team)
+	-- players without a team are counted as neutral
+	if not team or team == TEAM_NONE or TEAMS[team].alone then
+		return false
+	end
+
+	if GetConVar("ttt2_rolecheck_all_evil_roles"):GetBool() then
```

Comment:
That's something I definitely don't like. It differs the gameplay into two completely different ways. As well, it's a hidden option. Core functions and mechanics shouldn't be adjustable as players does not expect such changes. Additionally, I don't like the Evil approach, Imo, every role or better said, every team should decide on it's own whether it's "evil" -> And even that can't lead to a better handling, because every role interaction can be unique. That's why I said I'd like to modify things via hooks instead of role attributes.

TL;DR:
All roles except innocent are "evil" || "evil" == `IsSpecial()`

Reply:
We can leave the convar out. I just thought about the moaning players that want the default case back.

And we thought about defining an isEvil team flag, but we decided against it. We don't want to create a comiplicated system. A player is evil, if he isn't innocent. E.g. in a traitor tester not only trators should trigger it, but also necros, jackals etc. This was our intention.

Reply:
Yea, so we don't need an extra function but a simple inline check `Player:GetTeam() ~= TEAM_INNOCENT` :D That should be the most common case. Currently, I can't imagine a different behavior by different maps and if so, a convar wouldn't be the way to go, then it would need to be a map-based config

Reply:
If we don't have a convar, we don't need this function, yes. However we should also consider `TEAM_NONE` like we did. It is easier to read if both checks are inside a function

Reply:
Hmmm... That's a good point tho. Anyways, then please keep in mind that a team with `.alone` is still evil. Regarding:

```lua
if not team or team == TEAM_NONE or TEAMS[team].alone then
	return false
end
```

e.g. the SERIALKILLER has TEAM.alone

Reply:
ah-oh. Good to know. I didn't know what's the reason of `.alone`

Reply:
TL;DR: Teams with `.alone` should just be able to win with one (and only one) alive team player. As a result, if there is more than one Serialkiller, a WIN should just trigger if one of them died.

Reply:
I'd support leavin the convar out. I like doing it the modern way ^^